src/app/modules/core/core.errors.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * A custom base error class that encapsulates the name, message, status code,
+ * and logging meta string (if any) for the error.
+ */
+export class ApplicationError extends Error {
+  /**
+   * Http status code for the error to be returned in the response.
+   */
+  status: number
+  /**
+   * Meta string to be logged by the application logger, if any.
+   */
+  meta?: string
```

Comment:
This does not need to be a string too, can be changed any time; this is meant to be passed as-is to our winston logger.